src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -1012,6 +1014,30 @@ export class SearchInterface extends RootComponent implements IComponentBindings
         }
       });
     });
+    if (this.FacetDuplicate.length) {
+      this.logger.warn(
+        `The following facets have duplicate id/field:`,
+        this.FacetDuplicate,
+        `Ensure that each facet in your search interface has a unique id.`
+      );
+    }
+  }
+
+  private get FacetDuplicate(): Facet[] {
+    const duplicate: Facet[] = [];
+    const facets = ComponentsTypes.getAllFacetsInstance(this.root);
+    facets.forEach((facet: Facet) => {
+      facets.forEach((cmp: Facet) => {
+        if (facet == cmp) {
+          return;
+        }
+        if (facet.options.id === cmp.options.id) {
```

Comment:
Would this be okay if we had two facets, one configured with a field, and the other with an id equal to the first facet's field?

By default, when there is no id, the field becomes the id, but I am not sure if it is copied over to the `options.id` property. If yes, then this is good. If no, then the logic currently would not catch that case.

Reply:
just checked with these facets and I still have the warning
```
<div class="CoveoFacet" data-id="@filetype" data-title="FileType" data-field="@filetype" data-tab="All"></div>
<div class="CoveoFacet" data-title="FileType" data-field="@filetype" data-tab="All"></div>
<div class="CoveoFacet" data-title="Author" data-field="@author" data-depends-on="@filetype" data-tab="All">
```